src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbBySharedTable.java

```diff
@@ -605,6 +608,28 @@ public UpdateItemResult updateItem(UpdateItemRequest updateItemRequest) {
         return getAmazonDynamoDb().updateItem(updateItemRequest);
     }
 
+    @Override
+    public ListTablesResult listTables(String exclusiveStartTableName, Integer limit) {
+        if (getMtContext().getContextOpt().isEmpty()) {
+            // delegate to parent class' multitenant listTables call if no tenant is provided
+            return super.listTables(exclusiveStartTableName, limit);
+        } else {
+            TenantTable startKey = exclusiveStartTableName == null ? null :
+                new TenantTable(exclusiveStartTableName, getMtContext().getContext());
+            ListMetadataRequest listRequest =
+                new ListMetadataRequest().withLimit(limit).withExclusiveStartKey(startKey);
```

Comment:
Wouldn't it be more consistent with `MtDynamoDbTableDescriptionRepo#getTableDescription` to move the logic of computing the qualified key into the description repo?, i.e., have the repo implement `listTables(ListTablesRequest)`?

Reply:
makes sense, took a stab at doing that in this PR